lib/collection.js

```diff
@@ -1239,14 +1240,16 @@ Collection.prototype.createIndex = function(fieldOrSpec, options, callback) {
   if (typeof options === 'function') (callback = options), (options = {});
   options = options || {};
 
-  const createIndexOperation = new CreateIndexOperation(
-    this.s.db,
+  const prepared = prepareCreateIndex(fieldOrSpec, options);
```

Comment:
to me it looks like you can move the logic of `prepareCreateIndex` (which I think also could use a more descriptive name , maybe `makeIndexSpec`?) into the `CreateIndexesOperation` itself. You are always passing the user provided `options` to that operation, so it seems like you could do something like:

```
class CreateIndexOperation extends CommandOperationV2 {
  constructor(parent, collectionName, indexes, options) {
    super(parent, Array.isArray(indexes) ? options : makeSingleIndexSpecOptions(options));
    this.indexes = Array.isArray(indexes) ? indexes : [makeIndexSpec(indexes, options)];
    ...
   }
}
```   

We want to make sure that the `CreateIndexesOperation` is _always_ working with a well-defined index specification (something we would likely provide an interface for in TypeScript, for instance), but we need not clutter the rest of our codebase with implementation details I think.

Reply:
Yeah that was a good call, I moved all that logic inside `CreateIndexesOperation`. 👍